File: src/1298-Maximum-Candies-You-Can-Get-from-Boxes/1298.go
func maxCandies(status []int, candies []int, keys [][]int, containedBoxes [][]int, initialBoxes []int) int {
    q, res := initialBoxes, 0
    for len(q) > 0 {
        changed := false
        for q_len := len(q); q_len > 0; q_len-- {
            b := q[0]
            q = q[1:]
            if status[b] == 1 {
                changed = true
                res += candies[b]
                q = append(q, containedBoxes[b]...)
                for _, i := range keys[b] {
                    status[i] = 1
                }
            } else {
                q = append(q, b)
            }
        }
        if !changed {
            return res
        }
    }
    return res
}

File: src/1298-Maximum-Candies-You-Can-Get-from-Boxes/1298.py
class Solution:
    def maxCandies(self, status: List[int], candies: List[int], keys: List[List[int]], containedBoxes: List[List[int]], initialBoxes: List[int]) -> int:
        q, res = initialBoxes, 0
        while q:
            q_len, changed = len(q), False
            for _ in range(q_len):
                b = q.pop(0)
                if status[b]:
                    changed = True
                    res += candies[b]
                    q += containedBoxes[b]
                    for i in keys[b]:
                        status[i] = 1
                else:
                    q.append(b)

            if not changed:
                return res
        return res